lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/spatial/PointCloud.java

```diff
@@ -0,0 +1,317 @@
+         */
+        public byte getHitType() {
+            return hitType;
+        }
+
+        /**
+         * @return the distance to the origin of point cloud this point belongs to
+         */
+        public float getDistance() {
+            return distance;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Point other = (Point) o;
+            return super.equals(other) &&
+                    this.hitType == other.hitType &&
+                    Float.compare(this.distance, other.distance) == 0;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = super.hashCode();
+            result = 31 * result + Byte.hashCode(hitType);
+            result = 31 * result + Float.hashCode(distance);
+            return result;
+        }
+    }
+
+    public static class Absolute extends PointCloud {
+
+        private static final long serialVersionUID = 1L;
+
+        private transient List<Point> absoluteEndPointsWithHit = null;
+        private transient List<Point> relativeEndPoints = null;
+        private transient List<Point> relativeEndPointsWithHit = null;
+
+        public Absolute(RotationMatrix rotation, Vector3d reference, List<Point> absoluteEndPoints,
+                        long timestamp, double minRange, double maxRange) {
+            super(rotation, reference, absoluteEndPoints, timestamp, minRange, maxRange);
+        }
+
+        @Override
+        public List<Point> getAbsoluteEndPoints() {
+            return points;
+        }
+
+        @Override
+        public List<Point> getAbsoluteEndPointsWithHit() {
+            if (absoluteEndPointsWithHit == null) {
+                absoluteEndPointsWithHit = getAbsoluteEndPoints().stream().filter(Point::hasHit).collect(Collectors.toList());
+            }
+            return absoluteEndPointsWithHit;
+        }
+
+        @Override
+        public List<Point> getRelativeEndPoints() {
+            if (relativeEndPoints == null) {
+                relativeEndPoints = absoluteToRelative(points, p -> true);
+            }
+            return relativeEndPoints;
+        }
+
+        @Override
+        public List<Point> getRelativeEndPointsWithHit() {
+            if (relativeEndPointsWithHit == null) {
+                relativeEndPointsWithHit = absoluteToRelative(points, Point::hasHit);
+            }
+            return relativeEndPointsWithHit;
+        }
+
+        /**
+         * Transforms list of {@link Point}s in absolute coordinates to relative coordinates.         *
+         *
+         * @param absoluteEndPoints end points in absolute coordinates
+         * @return end points in relative coordinates
+         */
+        private List<Point> absoluteToRelative(List<Point> absoluteEndPoints, Predicate<Point> filter) {
```

Comment:
Aren't `absoluteToRelative(...)` and `relativeToAbsolute(...)` pretty much static? I don't yet see why we need to separate absolute (what does that even mean -> World Coordinates??) and relative point clouds if they can be freely converted anyways given a reference frame

Reply:
As explained above, yes we can freely convert them, but for performance reasons you want to get the best out of it depending on the use case (if you consume only absolute coordinates, then the sensor should already produce absolute coordinates, to avoid any cost-intensive conversion). It can be very time-consuming to do that for millions of points.